src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -96,17 +96,17 @@ namespace winrt::TerminalApp::implementation
             }
         });
 
-        // Focusing the ListView when the Command Palette control is set to Visible
-        // for the first time fails because the ListView hasn't finished loading by
-        // the time Focus is called. Luckily, We can listen to SizeChanged to know
-        // when the ListView has been measured out and is ready, and we'll immediately
-        // revoke the handler because we only needed to handle it once on initialization.
         _sizeChangedRevoker = _filteredActionsView().SizeChanged(winrt::auto_revoke, [this](auto /*s*/, auto /*e*/) {
-            // This does only fire once, when the size changes, which is the
-            // very first time it's opened. It does not fire for subsequent
-            // openings.
-
-            _sizeChangedRevoker.revoke();
+            // When we're in BottomUp mode, we need to adjust our own position
+            // so that our bottom is aligned with our origin. This will ensure
+            // that as the menu changes in size (as we filter results), the menu
+            // stays "attached" to the cursor.
+            if (Visibility() == Visibility::Visible && _direction == TerminalApp::SuggestionsDirection::BottomUp)
+            {
+                auto m = this->Margin();
+                m.Top = (_anchor.Y - ActualHeight());
+                this->Margin(m);
+            }
```

Comment:
Yea, I was hoping that by just starting with the right direction based off the size it opened with, then anything smaller would still make sense in that direction.

I worry that like, trying to rearrange the layout of the control while the size is changing is just gonna trigger another size change. I'm sure there's a _correct_ way to do this with the Arrange/Layout events in WinUI, but that's a dark part of the framework I've been trying my best to avoid 😅